lib/rules/space-infix-ops.js

```diff
@@ -75,26 +116,31 @@ module.exports = {
          * @private
          */
         function report(mainNode, culpritToken) {
+            const operator = culpritToken.value;
+            const must = spacesYes[operator] ? "must" : "must not";
+
             context.report({
                 node: mainNode,
                 loc: culpritToken.loc.start,
-                message: "Infix operators must be spaced.",
+                data: { must, operator },
+                message: "Operator `{{operator}}` {{must}} be surrounded by spaces.",
                 fix(fixer) {
+                    const shouldHaveSpaces = spacesYes[operator];
                     const previousToken = sourceCode.getTokenBefore(culpritToken);
-                    const afterToken = sourceCode.getTokenAfter(culpritToken);
-                    let fixString = "";
+                    const nextToken = sourceCode.getTokenAfter(culpritToken);
 
-                    if (culpritToken.range[0] - previousToken.range[1] === 0) {
-                        fixString = " ";
-                    }
+                    let fixString = sourceCode.getText(previousToken);
 
+                    if (shouldHaveSpaces) {
+                        fixString += " ";
+                    }
                     fixString += culpritToken.value;
-
-                    if (afterToken.range[0] - culpritToken.range[1] === 0) {
+                    if (shouldHaveSpaces) {
                         fixString += " ";
                     }
+                    fixString += sourceCode.getText(nextToken);
 
-                    return fixer.replaceText(culpritToken, fixString);
+                    return fixer.replaceText(mainNode, fixString);
```

Comment:
I think this fixer will be incorrect if a value contains multiple tokens. `fixString` only contains the text from the previous token and next token, but you're replacing the entire node. As a result,

```js
foo.bar+baz
```

will get fixed to:

```js
bar + baz
```

I would recommend using `fixer.replaceTextRange` instead, and only replacing the spacing and the operator, rather than replacing the tokens around it.

```js
const fixString = shouldHaveSpaces ? ` ${culpritToken.value} ` : culpritToken.value;

return fixer.replaceTextRange([previousToken.range[1], nextToken.range[0]], fixString);
```